### ISSUE TITLE
Bug: signed mutations are throwing an error.

### ISSUE THREAD

**Issue by magicspon** [jamesedmonston/graphql-authentication#103]
I'm having some issues with signed mutations requests.  I've not changed anything on my side, other than to upgrade to craft 4.

Using the latest version of all the things:

![image](https://user-images.githubusercontent.com/3268717/189951308-9ee3c55f-b06e-49ed-bef9-3a49237b7e71.png)

Unsigned (i.e. using the default gql headers)
![image](https://user-images.githubusercontent.com/3268717/189942692-50f7fb12-1939-4651-a4ac-b6c61bcd1ee0.png)

Signed with JWT header
```
{
  "errors": [
    {
      "debugMessage": "foreach() argument must be of type array|object, bool given",
      "message": "Internal server error",
      "extensions": {
        "category": "internal"
      },
      "file": "/home/forge/mysite.com/releases/869f1153148d27d0c1d7dbf5a4b7be2866d941cc/vendor/jamesedmonston/graphql-authentication/src/services/RestrictionService.php",
      "line": 291,
      "trace": [
        {
          "file": "/home/forge/mysite.com/releases/869f1153148d27d0c1d7dbf5a4b7be2866d941cc/vendor/craftcms/cms/src/web/ErrorHandler.php",
          "line": 79,
          "call": "yii\\base\\ErrorHandler::handleError()"
        },
        {
          "file": "/home/forge/mysite.com/releases/869f1153148d27d0c1d7dbf5a4b7be2866d941cc/vendor/jamesedmonston/graphql-authentication/src/services/RestrictionService.php",
          "line": 291,
          "call": "craft\\web\\ErrorHandler::handleError()"
        },
        {
          "file": "/home/forge/mysite.com/releases/869f1153148d27d0c1d7dbf5a4b7be2866d941cc/vendor/jamesedmonston/graphql-authentication/src/services/RestrictionService.php",
          "line": 66,
          "call": "jamesedmonston\\graphqlauthentication\\services\\RestrictionService::restrictMutationFields()"
        },
        {
          "call": "jamesedmonston\\graphqlauthentication\\services\\RestrictionService::jamesedmonston\\graphqlauthentication\\services\\{closure}()"
        },
        {
          "file": "/home/forge/mysite.com/releases/869f1153148d27d0c1d7dbf5a4b7be2866d941cc/vendor/yiisoft/yii2/base/Event.php",
          "line": 312,
          "function": "call_user_func()"
        },
        {
          "file": "/home/forge/mysite.com/releases/869f1153148d27d0c1d7dbf5a4b7be2866d941cc/vendor/yiisoft/yii2/base/Component.php",
          "line": 642,
          "call": "yii\\base\\Event::trigger()"
        },
        {
          "file": "/home/forge/mysite.com/releases/869f1153148d27d0c1d7dbf5a4b7be2866d941cc/vendor/craftcms/cms/src/base/Element.php",
          "line": 4610,
          "call": "yii\\base\\Component::trigger()"
        },
        {
          "file": "/home/forge/mysite.com/releases/869f1153148d27d0c1d7dbf5a4b7be2866d941cc/vendor/craftcms/cms/src/elements/Entry.php",
          "line": 1896,
          "call": "craft\\base\\Element::beforeSave()"
        },
        {
          "file": "/home/forge/mysite.com/releases/869f1153148d27d0c1d7dbf5a4b7be2866d941cc/vendor/craftcms/cms/src/services/Elements.php",
          "line": 2534,
          "call": "craft\\elements\\Entry::beforeSave()"
        },
        {
          "file": "/home/forge/mysite.com/releases/869f1153148d27d0c1d7dbf5a4b7be2866d941cc/vendor/craftcms/cms/src/services/Elements.php",
          "line": 795,
          "call": "craft\\services\\Elements::_saveElementInternal()"
        },
        {
          "file": "/home/forge/mysite.com/releases/869f1153148d27d0c1d7dbf5a4b7be2866d941cc/vendor/craftcms/cms/src/gql/base/ElementMutationResolver.php",
          "line": 163,
          "call": "craft\\services\\Elements::saveElement()"
        },
        {
          "file": "/home/forge/mysite.com/releases/869f1153148d27d0c1d7dbf5a4b7be2866d941cc/vendor/craftcms/cms/src/gql/resolvers/mutations/Entry.php",
          "line": 71,
          "call": "craft\\gql\\base\\ElementMutationResolver::saveElement()"
        },
        {
          "file": "/home/forge/mysite.com/releases/869f1153148d27d0c1d7dbf5a4b7be2866d941cc/vendor/webonyx/graphql-php/src/Executor/ReferenceExecutor.php",
          "line": 623,
          "call": "craft\\gql\\resolvers\\mutations\\Entry::saveEntry()"
        },
        {
          "file": "/home/forge/mysite.com/releases/869f1153148d27d0c1d7dbf5a4b7be2866d941cc/vendor/webonyx/graphql-php/src/Executor/ReferenceExecutor.php",
          "line": 550,
          "call": "GraphQL\\Executor\\ReferenceExecutor::resolveFieldValueOrError()"
        },
        {
          "file": "/home/forge/mysite.com/releases/869f1153148d27d0c1d7dbf5a4b7be2866d941cc/vendor/webonyx/graphql-php/src/Executor/ReferenceExecutor.php",
          "line": 474,
          "call": "GraphQL\\Executor\\ReferenceExecutor::resolveField()"
        },
        {
          "file": "/home/forge/mysite.com/releases/869f1153148d27d0c1d7dbf5a4b7be2866d941cc/vendor/webonyx/graphql-php/src/Executor/ReferenceExecutor.php",
          "line": 857,
          "call": "GraphQL\\Executor\\ReferenceExecutor::GraphQL\\Executor\\{closure}()"
        },
        {
          "call": "GraphQL\\Executor\\ReferenceExecutor::GraphQL\\Executor\\{closure}()"
        },
        {
          "file": "/home/forge/mysite.com/releases/869f1153148d27d0c1d7dbf5a4b7be2866d941cc/vendor/webonyx/graphql-php/src/Executor/ReferenceExecutor.php",
          "line": 859,
          "function": "array_reduce()"
        },
        {
          "file": "/home/forge/mysite.com/releases/869f1153148d27d0c1d7dbf5a4b7be2866d941cc/vendor/webonyx/graphql-php/src/Executor/ReferenceExecutor.php",
          "line": 490,
          "call": "GraphQL\\Executor\\ReferenceExecutor::promiseReduce()"
        },
        {
          "file": "/home/forge/mysite.com/releases/869f1153148d27d0c1d7dbf5a4b7be2866d941cc/vendor/webonyx/graphql-php/src/Executor/ReferenceExecutor.php",
          "line": 263,
          "call": "GraphQL\\Executor\\ReferenceExecutor::executeFieldsSerially()"
        },
        {
          "file": "/home/forge/mysite.com/releases/869f1153148d27d0c1d7dbf5a4b7be2866d941cc/vendor/webonyx/graphql-php/src/Executor/ReferenceExecutor.php",
          "line": 215,
          "call": "GraphQL\\Executor\\ReferenceExecutor::executeOperation()"
        },
        {
          "file": "/home/forge/mysite.com/releases/869f1153148d27d0c1d7dbf5a4b7be2866d941cc/vendor/webonyx/graphql-php/src/Executor/Executor.php",
          "line": 156,
          "call": "GraphQL\\Executor\\ReferenceExecutor::doExecute()"
        },
        {
          "file": "/home/forge/mysite.com/releases/869f1153148d27d0c1d7dbf5a4b7be2866d941cc/vendor/webonyx/graphql-php/src/GraphQL.php",
          "line": 162,
          "call": "GraphQL\\Executor\\Executor::promiseToExecute()"
        },
        {
          "file": "/home/forge/mysite.com/releases/869f1153148d27d0c1d7dbf5a4b7be2866d941cc/vendor/webonyx/graphql-php/src/GraphQL.php",
          "line": 94,
          "call": "GraphQL\\GraphQL::promiseToExecute()"
        },
        {
          "file": "/home/forge/mysite.com/releases/869f1153148d27d0c1d7dbf5a4b7be2866d941cc/vendor/craftcms/cms/src/services/Gql.php",
          "line": 513,
          "call": "GraphQL\\GraphQL::executeQuery()"
        },
        {
          "file": "/home/forge/mysite.com/releases/869f1153148d27d0c1d7dbf5a4b7be2866d941cc/vendor/jamesedmonston/graphql-authentication/src/services/TokenService.php",
          "line": 277,
          "call": "craft\\services\\Gql::executeQuery()"
        },
        {
          "call": "jamesedmonston\\graphqlauthentication\\services\\TokenService::setActiveSchema()"
        },
        {
          "file": "/home/forge/mysite.com/releases/869f1153148d27d0c1d7dbf5a4b7be2866d941cc/vendor/yiisoft/yii2/base/Event.php",
          "line": 312,
          "function": "call_user_func()"
        },
        {
          "file": "/home/forge/mysite.com/releases/869f1153148d27d0c1d7dbf5a4b7be2866d941cc/vendor/yiisoft/yii2/base/Component.php",
          "line": 642,
          "call": "yii\\base\\Event::trigger()"
        },
        {
          "file": "/home/forge/mysite.com/releases/869f1153148d27d0c1d7dbf5a4b7be2866d941cc/vendor/craftcms/cms/src/services/Gql.php",
          "line": 485,
          "call": "yii\\base\\Component::trigger()"
        },
        {
          "file": "/home/forge/mysite.com/releases/869f1153148d27d0c1d7dbf5a4b7be2866d941cc/vendor/craftcms/cms/src/controllers/GraphqlController.php",
          "line": 177,
          "call": "craft\\services\\Gql::executeQuery()"
        },
        {
          "call": "craft\\controllers\\GraphqlController::actionApi()"
        },
        {
          "file": "/home/forge/mysite.com/releases/869f1153148d27d0c1d7dbf5a4b7be2866d941cc/vendor/yiisoft/yii2/base/InlineAction.php",
          "line": 57,
          "function": "call_user_func_array()"
        },
        {
          "file": "/home/forge/mysite.com/releases/869f1153148d27d0c1d7dbf5a4b7be2866d941cc/vendor/yiisoft/yii2/base/Controller.php",
          "line": 178,
          "call": "yii\\base\\InlineAction::runWithParams()"
        },
        {
          "file": "/home/forge/mysite.com/releases/869f1153148d27d0c1d7dbf5a4b7be2866d941cc/vendor/yiisoft/yii2/base/Module.php",
          "line": 552,
          "call": "yii\\base\\Controller::runAction()"
        },
        {
          "file": "/home/forge/mysite.com/releases/869f1153148d27d0c1d7dbf5a4b7be2866d941cc/vendor/craftcms/cms/src/web/Application.php",
          "line": 301,
          "call": "yii\\base\\Module::runAction()"
        },
        {
          "file": "/home/forge/mysite.com/releases/869f1153148d27d0c1d7dbf5a4b7be2866d941cc/vendor/yiisoft/yii2/web/Application.php",
          "line": 103,
          "call": "craft\\web\\Application::runAction()"
        },
        {
          "file": "/home/forge/mysite.com/releases/869f1153148d27d0c1d7dbf5a4b7be2866d941cc/vendor/craftcms/cms/src/web/Application.php",
          "line": 286,
          "call": "yii\\web\\Application::handleRequest()"
        },
        {
          "file": "/home/forge/mysite.com/releases/869f1153148d27d0c1d7dbf5a4b7be2866d941cc/vendor/yiisoft/yii2/base/Application.php",
          "line": 384,
          "call": "craft\\web\\Application::handleRequest()"
        },
        {
          "file": "/home/forge/mysite.com/releases/869f1153148d27d0c1d7dbf5a4b7be2866d941cc/web/index.php",
          "line": 12,
          "call": "yii\\base\\Application::run()"
        }
      ]
    }
  ],
  "data": {
    "campaign": null
  }
}
```

**Comment by cvitko** [comment]
Just a guess. Account param is typed as Array of Integers and you are passing just an Integer value.

**Comment by magicspon** [comment]
I'll have a look.  That wouldn't explain why it works using the standard graphql token.

**Comment by jamesedmonston** [comment]
@magicspon looks like this is an issue with the plugin stepping in and trying to validate mutations on custom elements (in this case, coming from a plugin) – I'll add in some additional checks to ensure it only validates first-party entries and assets.

**Comment by jamesedmonston** [comment]
I totally misread your mutation – I thought it was a mutation coming from a plugin.

Okay, cool. I'll try and take a look this evening.

**Comment by jamesedmonston** [comment]
@magicspon fixed in `2.1.2`
